src/QsCompiler/Compiler/Templates/CppInterop.cs

```diff
@@ -0,0 +1,193 @@
+                    DataType.RangeType => $"Range array",
+                    DataType.ResultType => $"Result array",
+                    DataType.StringType => $"String array",
+                    _ => throw new NotSupportedException($"Unsupported array type {this.Type}")
+                },
+                _ => throw new NotSupportedException($"Unsupported argument type {this.Type}")
+            };
+        }
+
+        public string CliDescription() => $"A {this.CliTypeDescription()} value for the {this.Name} argument";
+
+        public string? CppVarType()
+        {
+            return this.Type switch
+            {
+                DataType.BoolType => "char",
+                DataType.IntegerType => "int64_t",
+                DataType.DoubleType => "double_t",
+                DataType.PauliType => "PauliId",
+                DataType.RangeType => "RangeTuple",
+                DataType.ResultType => "char",
+                DataType.StringType => "string",
+                DataType.ArrayType => this.ArrayType switch
+                {
+                    DataType.BoolType => "vector<char>",
+                    DataType.IntegerType => "vector<int64_t>",
+                    DataType.DoubleType => "vector<double_t>",
+                    DataType.PauliType => "std::vector<PauliId>",
+                    DataType.RangeType => "vector<RangeTuple>",
+                    DataType.ResultType => "vector<char>",
+                    DataType.StringType => "vector<string>",
+                    _ => throw new NotSupportedException($"Unsupported array type {this.Type}")
+                },
+                _ => throw new NotSupportedException($"Unsupported argument type {this.Type}")
+            };
+        }
+
+        public string? CppVarInitialValue()
+        {
+            return this.Type switch
+            {
+                DataType.BoolType => "InteropFalseAsChar",
+                DataType.IntegerType => "0",
+                DataType.DoubleType => "0.0",
+                DataType.PauliType => "PauliId::PauliId_I",
+                DataType.RangeType => null,
+                DataType.ResultType => "InteropResultZeroAsChar",
+                DataType.StringType => null,
+                DataType.ArrayType => this.ArrayType switch
+                {
+                    DataType.BoolType => null,
+                    DataType.IntegerType => null,
+                    DataType.DoubleType => null,
+                    DataType.PauliType => null,
+                    DataType.RangeType => null,
+                    DataType.ResultType => null,
+                    DataType.StringType => null,
+                    _ => throw new NotSupportedException($"Unsupported array type {this.Type}")
+                },
+                _ => throw new NotSupportedException($"Unsupported argument type {this.Type}")
+            };
+        }
+
+        public string? TransformationType()
```

Comment:
Looking at the usage of this in the template, a slightly more descriptive name for this method would be `TransformerMapName`.

To avoid possible mismatches with the names given to the maps in the template, creating a `static` method called `DataTypeTransformerMapName`  that performs this operation and using it in the class method and in the template would help.

Reply:
done